### PR TITLE
Loggers

### PR DESCRIPTION
**Pull request by MRossol**
@grantbuster and @kdheepak
I ran into some funky ass logging issue when refactoring reVX for CLI tests so I decided to take a swing and cleaning up our logging methodology namely with a focus on fixing duplicate logging.

Changes here are:
1) In setup_logger check to see if the log_dir exists, if not warn the user and add a stream handler instead
2) Make it possible to setup init_multi without a log dir
3) Fix the handler checking logic. Now each handler has a name (stream or the log file) if a handler with the same name exists a new handler is not added, BUT the log level is checked and increased if needed
4) This was the biggest and hairiest one that I could take or leave. There is now logic to remove any children loggers from the LOGGERS registry and to remove any handlers from children loggers if a parent logger is initialized. 
- The upside here is that it prevents duplicate logging every time
- The downside is that the log levels get a bit confusing. I had to remove the default logger.setLevel("DEBUG") because all children always logged debug messages, but I think I was seeing this earlier too. 

In a nutshell logging propagation is a blessing and a curse...

### CODE REVIEW

**Comment by grantbuster** [comment]
@MRossol you've been busy! looks good but im curious, have you tried running reV to see if there are any unexpected logging behaviors when running from pipeline?

**Comment by MRossol** [comment]
Yeah, this was a HUGE rabbit hole, it all started because of some funky logging behavior in the reVX tests... I haven't run the full-scale test but can, what sort of behavior are you worried about?

**Comment by grantbuster** [comment]
I'm not super concerned just hopeful that this shouldnt affect the expected reV logging usage much / at all?

**Comment by MRossol** [comment]
That is the hope. I will admit that I was seeing some strange behavior with debug logs showing up when not requested, but I was seeing that before this refactor AND I was seeing some annoying duplicate logging beforehand which will definitely be fixed! Running the full pipeline test now.

**Comment by MRossol** [comment]
@grantbuster, mind taking a look at the full test logs, they look good to my (not as well trained) eyes...
